### PR TITLE
add plot manager and several physics parameters calculation

### PR DESCRIPTION
**Pull request by j8xixo12**
In this PR added following physics parameters calculation:
1. Temperature
2. Entopy
3. Internal energy

Also added plot manager which can save current plot via file explorer window and it can switch plot layout between
grid layout or single plot by click button. 

Due to plot manager inherit from `PySide6.QWidgets.QWidget`, `PySide6` is mandatory now. 

Connect to #202

### CODE REVIEW

File: cpp/modmesh/onedim/pymod/wrap_onedim.cpp [hunk 1]
```diff
@@ -88,7 +88,19 @@ class MODMESH_PYTHON_WRAPPER_VISIBILITY WrapEuler1DCore
...
                 "pressure",
                 [](wrapped_type & self)
-                { return to_ndarray(self.pressure()); });
+                { return to_ndarray(self.pressure()); })
+            .def_property_readonly(
+                "temperature",
+                [](wrapped_type & self)
+                { return to_ndarray(self.temperature()); })
+            .def_property_readonly(
+                "int_energy",
```
Comment (yungyuc) [inline cpp/modmesh/onedim/pymod/wrap_onedim.cpp]: Please spell out the full name "internal_energy"

File: modmesh/app/euler1d.py [hunk 2]
```diff
@@ -61,73 +66,315 @@ def load_app():
...
     unit: str = ""
 
+    def bind_axis(self, axis):
+        self.axis = axis
+
     def update(self, adata, ndata):
         self.ana.set_ydata(adata.copy())
         self.num.set_ydata(ndata.copy())
+        self.axis.relim()
+        self.axis.autoscale_view()
```
Comment (yungyuc) [inline modmesh/app/euler1d.py:79]: Sometimes we do not want the plot to autoscale.  It's OK to use it now but we should plan for adding an option to control the behavior in the future.

File: modmesh/app/euler1d.py [hunk 3]
```diff
@@ -61,73 +66,315 @@ def load_app():
...
+                            wspace=0.2, hspace=0.2)
 
-    def update_lines(self, shocktube):
-        self.density.update(adata=shocktube.density_field,
-                            ndata=shocktube.svr.density[::2])
-        self.pressure.update(adata=shocktube.pressure_field,
-                             ndata=shocktube.svr.pressure[::2])
-        self.velocity.update(adata=shocktube.velocity_field,
-                             ndata=shocktube.svr.velocity[::2])
+        frameContainer.addWidget(canvas)
```
Comment (yungyuc) [inline modmesh/app/euler1d.py:371]: We are mixing `camelCase` and `snake_case` conventions.  We can live with it now.  After we have more code and experience we should come up with a way to reconcile the conflict in styles.

File: modmesh/app/euler1d.py [hunk 4]
```diff
@@ -61,73 +66,315 @@ def load_app():
...
+            if isinstance(item, QLayout):
+                self.delete_layout(item)
+            else:
+                item.widget().setParent(None)
+                item.widget().deleteLater()
+
+        self.graphLayout.deleteLater()
+        self.graphContainer.update()
+
+        self.use_gridLayout = not self.use_gridLayout
```
Comment (yungyuc) [inline modmesh/app/euler1d.py]: I think we should not use both `snake_camelCase` style in one name.  Could you please change it to either `snake_case` or `camelCase`?

File: modmesh/app/euler1d.py [hunk 5]
```diff
@@ -61,73 +66,315 @@ def load_app():
...
 
 
-class Plot:
-    def __init__(self, figsize=(15, 10)):
-        self.fig = Figure(figsize=figsize)
-        self.canvas = FigureCanvas(self.fig)
-        self.ax = self.canvas.figure.subplots()
-        self.ax.set_ylim(-0.2, 1.2)
-        self.ax.grid()
+class PlotManager(QWidget):
```
Comment (yungyuc) [inline modmesh/app/euler1d.py:84]: By doing this (and the module-level importing pyside6), we are requiring pyside6 in this app.  The PR and check-in log should make this new requirement explicit.

**Review by yungyuc — CHANGES_REQUESTED** [review]

File: cpp/modmesh/onedim/Euler1DCore.hpp [hunk 6]
```diff
@@ -107,12 +108,18 @@ class Euler1DCore
     SimpleArray<double> const & gamma() const { return m_gamma; }
     SimpleArray<double> & gamma() { return m_gamma; }
 
-    double density(size_t it) const { return m_so0(it, 0); }
+    inline double density(size_t it) const { return m_so0(it, 0); }
```
Comment (yungyuc) [inline cpp/modmesh/onedim/Euler1DCore.hpp]: Please remove the inlines.  They are not effective.  Code in the class declaration is automatically inline.

File: cpp/modmesh/onedim/Euler1DCore.hpp [hunk 7]
```diff
@@ -323,4 +330,4 @@ inline void Euler1DCore::march_alpha(size_t steps)
 }
 
 } /* end namespace onedim */
-} /* end namespace modmesh */
\ No newline at end of file
+} /* end namespace modmesh */
```
Comment (yungyuc) [inline cpp/modmesh/onedim/Euler1DCore.hpp]: Please avoid the unnecessary change of new line or end of file.

Comment (j8xixo12) [reply cpp/modmesh/onedim/Euler1DCore.hpp]: I remove unnecessary change int https://github.com/solvcon/modmesh/pull/210/commits/a67a8e26dde8c60e3cae3053c7232cac12f9acf4

File: cpp/modmesh/onedim/Euler1DCore.hpp (hunk 6 above)
Comment (j8xixo12) [reply cpp/modmesh/onedim/Euler1DCore.hpp]: I remove these unnecessary inline in https://github.com/solvcon/modmesh/pull/210/commits/c2fc36b09a7ae9a89586d7623e7f75454ac36a6e

File: cpp/modmesh/onedim/pymod/wrap_onedim.cpp (hunk 1 above)
Comment (j8xixo12) [reply cpp/modmesh/onedim/pymod/wrap_onedim.cpp]: I fixed it in https://github.com/solvcon/modmesh/pull/210/commits/ec805433b0646bc4dada82157bf5492707291ea1

File: modmesh/app/euler1d.py (hunk 3 above)
Comment (j8xixo12) [reply modmesh/app/euler1d.py:371]: I fixed it in https://github.com/solvcon/modmesh/pull/210/commits/ec805433b0646bc4dada82157bf5492707291ea1

File: modmesh/app/euler1d.py (hunk 4 above)
Comment (j8xixo12) [reply modmesh/app/euler1d.py]: ditto.

**Comment by j8xixo12** [comment]
I'm doing some stupid, I'll fix it.

File: cpp/modmesh/onedim/Euler1DCore.hpp [hunk 8]
```diff
@@ -111,8 +112,14 @@ class Euler1DCore
     SimpleArray<double> density() const;
     double velocity(size_t it) const { return m_so0(it, 1) / (m_so0(it, 0) + TINY); }
     SimpleArray<double> velocity() const;
-    double pressure(size_t it) const;
+    inline double pressure(size_t it) const;
```
Comment (yungyuc) [inline cpp/modmesh/onedim/Euler1DCore.hpp]: The inline should be placed at the definition, not the declaration, IIUC.

File: modmesh/app/euler1d.py (hunk 5 above)
Comment (yungyuc) [reply modmesh/app/euler1d.py:84]: @j8xixo12 could you please address this point as well?  Quote:

> By doing this (and the module-level importing pyside6), we are requiring pyside6 in this app. The PR and check-in log should make this new requirement explicit.

File: modmesh/app/euler1d.py (hunk 4 above)
Comment (yungyuc) [reply modmesh/app/euler1d.py]: I am not seeing it fixed in the review.

**Review by yungyuc — CHANGES_REQUESTED** [review]

File: cpp/modmesh/onedim/Euler1DCore.hpp (hunk 8 above)
Comment (j8xixo12) [reply cpp/modmesh/onedim/Euler1DCore.hpp]: Yes, I correct the `inline` specifier usage in https://github.com/solvcon/modmesh/pull/210/commits/d3348fedef5a2385dd8e07995f9057f2d9e4a8bb

File: modmesh/app/euler1d.py (hunk 4 above)
Comment (j8xixo12) [reply modmesh/app/euler1d.py]: I fixed it in https://github.com/solvcon/modmesh/pull/210/commits/b9ea25486ed643d5ee0974dd8d17c1f22f4285e8

File: modmesh/app/euler1d.py (hunk 5 above)
Comment (j8xixo12) [reply modmesh/app/euler1d.py:84]: I just updated PR message to remind `PySide6` is mandatory now.
Since `PySide6` is mandatory, the `MODMESH_USE_PYSIDE` seems is unnecessary, but it seems some issue need to fix https://github.com/solvcon/modmesh/blob/965aff9ca52ab3f00f80e4ce32aa48adbfbf2038/modmesh/app/euler1d.py#L153

I’ll try to reslove it.

Comment (yungyuc) [reply modmesh/app/euler1d.py:84]: If the issue does not get in your way of this PR, it's OK to use another PR to fix it.

**Review by yungyuc — APPROVED** [review]
Can you squash the commits?  I am fine to have a couple of commits but 14 (the current number) is unnecessarily many.  Please leave a comment when you finish squashing and would like the PR to be merged.

**Comment by j8xixo12** [comment]
I just squash all commits.

File: modmesh/app/euler1d.py [hunk 9]
```diff
@@ -34,10 +34,15 @@
 import numpy as np
 
 import matplotlib
-from matplotlib.backends.qt_compat import QtWidgets
+import matplotlib.pyplot
 from matplotlib.backends.backend_qtagg import (
-    FigureCanvas, NavigationToolbar2QT as NavigationToolbar)
+    FigureCanvas)
```
Comment (yungyuc) [inline modmesh/app/euler1d.py]: This doesn't need a new line.  Please keep it in the same line as above.

**Review by yungyuc — CHANGES_REQUESTED** [review]
The checkin log in the squashed commit is not clear.  Please improve the wording and make sure you mention the requirement of pyside6.

File: modmesh/app/euler1d.py (hunk 9 above)
Comment (j8xixo12) [reply modmesh/app/euler1d.py]: Fix it in https://github.com/solvcon/modmesh/pull/210/commits/214b4743bae1995cf7c451855880a397eae559fc

**Comment by j8xixo12** [comment]
> The checkin log in the squashed commit is not clear. Please improve the wording and make sure you mention the requirement of pyside6.

I rewrite the commit message to make it shorter and easily to understand!
I also remind `PySide6` is mandatory in `euler1d` app.

**Review by yungyuc — APPROVED** [review]
LGTM
